serial_linux.go

```diff
@@ -62,10 +62,15 @@ func openPort(name string, baud int) (rwc io.ReadWriteCloser, err error) {
 	}()
 
 	fd := f.Fd()
+	// set blocking / non-blocking read
+	var minBytesToRead uint8 = 1
+	if nonBlockingRead == true {
+		minBytesToRead = 0
+	}
 	t := syscall.Termios{
 		Iflag:  syscall.IGNPAR,
 		Cflag:  syscall.CS8 | syscall.CREAD | syscall.CLOCAL | rate,
-		Cc:     [32]uint8{syscall.VMIN: 1},
+		Cc:     [32]uint8{syscall.VMIN: minBytesToRead, syscall.VTIME: uint8(readTimeout)},
```

Comment:
Converting readTimeout to a uint8 will silently wrap values larger than 25.5 seconds.  Can you test for that case instead and return an error if the requested timeout is too long?